indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/encoding/Asn1Decoding.kt

```diff
@@ -54,7 +81,7 @@ private class Asn1Reader(input: ByteArray) {
             } else if (tlv.tag.isConstructed) { //custom tags, we don't know if it is a SET OF, SET, SEQUENCE,… so we default to sequence semantics
                 result.add(Asn1CustomStructure(Asn1Reader(tlv.content).doParse(), tlv.tag.tagValue, tlv.tagClass))
             } else result.add(Asn1Primitive(tlv.tag, tlv.content))
-
+            if (single) return result
```

Comment:
this feels like a really ugly workaround; surely `doParseOne` should be the function (read + decode + return single asn1elm), and then `doParseMany` should keep calling `doParseOne` while `rest.isNotEmpty()`, and collecting a list?

Reply:
invoking it toplevel is a special case that you never need anywhere else. this adds the least complexity

Reply:
i don't see any downside to my proposition, though; and it's far cleaner

Reply:
done